src/components/NewsFeed/MakePost.jsx

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import "./MakePost.css";
+import PropTypes from "prop-types";
+import firebaseWrapper from "../../firebaseWrapper";
+
+class MakePost extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      text: "",
+      photo: null,
+      likes: 0
+    };
+  }
+
+  textInputHandler = event => {
+    this.setState({ text: event.target.value });
+  };
+
+  fileUploadHandler = () => {
+    const pic = document.getElementById("fileItem").files[0];
+    this.setState({ photo: pic });
+  };
+
+  postSubmitHandler = event => {
+    const { userId } = this.props;
+    const { photo, text, likes } = this.state;
+    const resetState = () => {
+      this.setState({ text: "", photo: null, likes: 0 });
+    };
+    event.preventDefault();
+    if (photo !== null) {
+      firebaseWrapper.sendPostWithPhoto(userId, text, likes, photo, resetState);
+    } else {
+      firebaseWrapper.sendPost(userId, text, likes);
```

Comment:
sendPost should also take in a callback. You can probably combine the two `sendPost` functions. The reason you want to do this is because you want to reset the state AFTER firebase gets the updates. Doing it the way you're doing now does not have this guarantee.